tests/cross_functional/system_test/test_clone_deletion_after_max_cluster_space_utilization.py

```diff
@@ -0,0 +1,214 @@
+        )
+        clones_list = []
+
+        if interface_type == constants.CEPHBLOCKPOOL:
+            for clone_num in range(self.num_of_clones + 1):
+                logger.info(f"Start creation of clone number {clone_num}.")
+                cloned_pvc_obj = pvc_clone_factory(
+                    self.pvc_obj, storageclass=self.pvc_obj.backed_sc, timeout=600
+                )
+                cloned_pvc_obj.reload()
+
+                # flatten the image
+                self.flatten_image(cloned_pvc_obj)
+
+                clones_list.append(cloned_pvc_obj)
+                logger.info(
+                    f"Clone with name {cloned_pvc_obj.name} for {self.pvc_size} pvc {self.pvc_obj.name} was created."
+                )
+
+        else:
+            for clone_num in range(self.num_of_clones + 1):
+                logger.info(f"Start creation of clone number {clone_num}.")
+                cloned_pvc_obj = pvc_clone_factory(
+                    self.pvc_obj, storageclass=self.pvc_obj.backed_sc, timeout=600
+                )
+                cloned_pvc_obj.reload()
+                clones_list.append(cloned_pvc_obj)
+                logger.info(
+                    f"Clone with name {cloned_pvc_obj.name} for {self.pvc_size} pvc {self.pvc_obj.name} was created."
+                )
+
+        logger.info(
+            "Verify 'CephClusterCriticallyFull' ,CephOSDNearFull Alerts are seen "
+        )
+        expected_alerts = ["CephOSDNearFull", "CephOSDCriticallyFull"]
+        prometheus = PrometheusAPI(threading_lock=threading_lock)
+        sample = TimeoutSampler(
+            timeout=600,
+            sleep=10,
+            func=prometheus.verify_alerts_via_prometheus,
+            expected_alerts=expected_alerts,
+            threading_lock=threading_lock,
+        )
+        if not sample.wait_for_func_status(result=True):
+            logger.error(f"The alerts {expected_alerts} do not exist after 1200 sec")
+            raise TimeoutExpiredError
+
+        # Make the cluster out of full by increasing the full ratio.
+        logger.info("Change Ceph full_ratio from from 85% to 95%")
+
+        change_ceph_full_ratio(95)
+        # After the cluster is out of full state and IOs started , Try to delete clones.
+        # Delete the clones one by one and wait for deletion
+        logger.info(
+            f"Start deleting {self.num_of_clones} clones on {interface_type} PVC of size {self.pvc_size} Gi."
+        )
+
+        for index, clone in enumerate(clones_list):
+            index += 1
+            pvc_reclaim_policy = clone.reclaim_policy
+            clone.delete()
+            logger.info(f"Deletion of the clone name is {clone.name}.")
+            clone.ocp.wait_for_delete(clone.name, self.timeout)
+            if pvc_reclaim_policy == constants.RECLAIM_POLICY_DELETE:
+                helpers.validate_pv_delete(clone.backed_pv)
```

Comment:
you can actually delete the clones in parallel, instead of deleting them in serial manner. That can help us catch any race conditions and as well as speed up the test execution time. This comment is not a blocker for this PR but can be taken as an enhancement in your next PR.